assets/js/translator.js

```diff
@@ -447,12 +447,35 @@ function getPairs() /*: JQueryPromise<any> */ {
             }
         });
 
-        for(var srcLang in pairs) {
-            // Default for new users is first available pair
-            curSrcLang = srcLang;
-            curDstLang = pairs[srcLang][0];
-            break;
+        // Default for new users is first available Browser Preference Language pair
+        var browserLangs = navigator.languages; // Chrome, Mozilla and Safari
+        var prefLang;
+        $.each(browserLangs, function(i, langVal) {
+            var browserLang = getLangCode(langVal);
+            if(checkLangPairAvailable(browserLang)) {
+                prefLang = browserLang;
+                return false;
+            }
+        });
+        if(typeof prefLang === 'undefined') {
+            if(navigator.userlanguage !== undefined || navigator.browserlanguage !== undefined) {
+                var ieLang = getLangCode(navigator.userlanguage || navigator.browserlanguage); // Internet Explorer
+                if(checkLangPairAvailable(ieLang)) {
+                    curSrcLang = iso639CodesInverse[ieLang];
+                }
+                else {
+                    noPrefSrcLang();
+                }
+            }
+            else {
+                noPrefSrcLang();
+            }
+        }
+        else {
+            curSrcLang = iso639CodesInverse[prefLang];
```

Comment:
Instead, set `prefSrcLang`. Then, set `curSrcLang` once at the end if you have `prefSrcLang`.

Reply:
Done :)